### PR TITLE
Renaming hand to controller in input simulation related files

### PR DESCRIPTION
**Pull request by MaxWang-MS**
## Overview
This is the first of a set of PRs that will eventually add motion controller simulation and test support to MRTK. This PR specifically focuses on making name changes so that simulation is no longer confined to just hands but applies to controllers in general.

## Changes
- Change reference to "hand" to a more generic term "controller" to get ready for the upcoming support of simulated controller in editor play mode and tests.

### CODE REVIEW

File: Assets/MRTK/Core/Providers/InputSimulation/InputSimulationEnum.cs [hunk 1]
```diff
@@ -43,25 +44,58 @@ public enum EyeGazeSimulationMode
...
+        /// </summary>
+        MotionController,
+    }
+
+    /// <summary>
+    /// Defines for how input simulation handles controllers
+    /// </summary>
+    [Obsolete("Use ControllerSimulationMode instead.")]
+    public enum HandSimulationMode
+    {
```
Comment (CDiaz-MS) [inline Assets/MRTK/Core/Providers/InputSimulation/InputSimulationEnum.cs:78]: nit: add a region Obsolete Properties for consistency with the other files

File: Assets/MRTK/Core/Providers/InputSimulation/BaseInputSimulationService.cs [hunk 2]
```diff
@@ -66,66 +67,74 @@ public override IMixedRealityController[] GetActiveControllers()
...
+                    
+                }
             }
+            
+            RemoveControllerDevice(handedness);
+            
         }
 
-        public SimulatedHand GetHandDevice(Handedness handedness)
+        public BaseController GetControllerDevice(Handedness handedness)
```
Comment (CDiaz-MS) [inline Assets/MRTK/Core/Providers/InputSimulation/BaseInputSimulationService.cs:89]: We probably need a section in the release notes that explain the reason for the breaking changes.

The notes would be added here: https://microsoft.github.io/MixedRealityToolkit-Unity/Documentation/ReleaseNotes.html#breaking-changes

File: Assets/MRTK/Core/Providers/InputSimulation/InputSimulationEnum.cs (hunk 1 above)
Comment (MaxWang-MS) [reply Assets/MRTK/Core/Providers/InputSimulation/InputSimulationEnum.cs]: Fixed!

File: Assets/MRTK/Core/Providers/InputSimulation/BaseInputSimulationService.cs (hunk 2 above)
Comment (MaxWang-MS) [reply Assets/MRTK/Core/Providers/InputSimulation/BaseInputSimulationService.cs:89]: Agreed, but if you don't mind I want to add the section in the next PR as things will be clearer there.

Comment (MaxWang-MS) [reply Assets/MRTK/Core/Providers/InputSimulation/BaseInputSimulationService.cs:89]: @CDiaz-MS Just realized that this is actually not a breaking change as the original function with the same signature was moved down to the obsolete section. But I will still mention this in the next PR.

**Pull request merged**